### DISCUSSION TITLE
Tor exitnode blacklist

### DISCUSSION THREAD

**Discussion by karolyi** [wdes/security#247] [Ideas]
Hey,

kinda wanted to transfer this to your gitea (so basically, off-github), but I don't (yet) have an account there, and it seems to be acting slow.

So here's the idea: tor exit node list. Blocking Tor exit nodes is a controversial topic:

- the argument against it is, you block dissenters and people who aren't aligned with "the narrative"
- the argument for it is, a lot of abuse comes from those exit nodes.

I started using the collection and blockage of those exit nodes since they abused one of my services.

Here's the source, it relies on somebody else's collection of exit nodes (https://www.dan.me.uk/tornodes) which can be only pulled every half an hour:

https://git.ksol.io/karolyi/forum-django/src/branch/master/backend/forum/utils/access_governor/ip_collector/tor.py

Feel free to integrate it in any way you like.

**Comment by williamdes** [comment]
Feel free to continue using GitHub
I think I will advertise contributions to the main repository using GitHub or sending patches by email
Or what the user wants, maybe another gitea or any stuff I can clone and merge
